### PR TITLE
Fix argmax for ChainedVector with function 1st arg

### LINKED ISSUE #51: argmax for chained vector
@quinnj see the discussion in https://discourse.julialang.org/t/findmax-and-friends-confusing-behaviour-to-be-introduced-in-1-7/61904/79

I think the referenced definition should be fixed. Right?

### PR DESCRIPTION
**Pull request by quinnj**
Fixes #51.

### CODE REVIEW

**Review by bkamins — CHANGES_REQUESTED** [review]
I think it is not the point of this change, see the example on Julia 1.7:
```
julia> argmax(abs, -10:5)
-10
```
so a value from domain is returned. With your fix you would return a value from co-domain:
```
julia> argmax(abs, ChainedVector([-10:5])) # this is after your change
10
```
and before your change we have:
```
```
julia> argmax(abs, ChainedVector([-10:5])) # this is before your change
1
```

**Comment by quinnj** [comment]
Ah, now it makes sense; I kept feeling I was just not getting this whole issue; thanks for the example. The code now should be correct.

File: src/chainedvector.jl [hunk 1]
```diff
@@ -767,10 +767,10 @@ end
...
 Base.findmax(x::ChainedVector) = findmax(identity, x)
 Base.findmin(x::ChainedVector) = findmin(identity, x)
-Base.argmax(x::ChainedVector) = findmax(identity, x)[2]
-Base.argmin(x::ChainedVector) = findmin(identity, x)[2]
-Base.argmax(f::F, x::ChainedVector) where {F} = findmax(f, x)[2]
-Base.argmin(f::F, x::ChainedVector) where {F} = findmin(f, x)[2]
+Base.argmax(x::ChainedVector) = findmax(identity, x)[1]
+Base.argmin(x::ChainedVector) = findmin(identity, x)[1]
+Base.argmax(f::F, x::ChainedVector) where {F} = x[findmax(f, x)[2]]
+Base.argmin(f::F, x::ChainedVector) where {F} = x[findmin(f, x)[2]]
```
Comment (nickrobinson251) [inline src/chainedvector.jl:773]: why doesn't Base have these definitions? i.e. just forward `argmin` -> `findmin` (at least for AbstractArrays)

Comment (bkamins) [reply src/chainedvector.jl:773]: I think the reason is that in Base `domain` can be more general, and in particular do not implement `findmax`.

**Review by bkamins — APPROVED** [review]
looks good now

File: src/chainedvector.jl (hunk 1 above)
Comment (nickrobinson251) [reply src/chainedvector.jl:773]: interesting, i'll follow-up on the Julia repo or slack

it looks to me like Base has a default `findmax(A::AbstractArray)` method, and also that an `AbstractArray` is required to implement `getindex`, and these are the only things needed to get `argmax` for free :) 

(anyway, this was just a curiosity -- PR looks good to me too!)

Comment (nickrobinson251) [reply src/chainedvector.jl:773]: oh, i was looking at pre-1.7 Base code, i see how it's changed now -- thanks for the discussion :)

**Comment by quinnj** [comment]
We match Base's behavior here?
```julia
julia> a = collect(20:40)
21-element Vector{Int64}:
 20
 21
 22
 23
 24
 25
 26
 27
 28
 29
 30
 31
 32
 33
 34
 35
 36
 37
 38
 39
 40

julia> argmax(a; dims=(1,))
1-element Vector{Int64}:
 21

julia> b = ChainedVector([a])
21-element ChainedVector{Int64, Vector{Int64}}:
 20
 21
 22
 23
 24
 25
 26
 27
 28
 29
 30
 31
 32
 33
 34
 35
 36
 37
 38
 39
 40

julia> argmax(b; dims=(1,))
1-element Vector{Int64}:
 21
```

**Pull request merged**